### DISCUSSION TITLE
User defined host to connect the requested URL in input file.

### DISCUSSION THREAD

**Discussion by ehsandeep** [projectdiscovery/nuclei#1473] [Ideas]
- [ ] Supporting `-src-ip` flag to predefine `IP` / `hostname` to connect the requested URL in input file.

`-src-ip` this flag is to indicate that the input file expect `hostName`, starting with `hostURL` as following:-

```
https://hostURL,hostName
```
```
https://a.google.com,1.1.1.1
https://x.google.com,1.2.1.1
https://a.google.com,1.1.1.1
https://s.google.com,1.1.3.1
https://1.1.1.1, subs.google.com
https://2.2.2.2, subs2.google.com
```

Notes:- 

- 1st input is always going to be `requesting URL`, and 2nd input will be used as `hostName` for the request.
- `-src-ip` flag only works `http` module.
- Input de-duplication should check for `https://hostURL,hostName` and not only `https://hostURL`.
- Handling space in the input file in-between `https://hostURL` and `hostName`, eg:- `https://2.2.2.2, subs2.google.com`

**Comment by ehsandeep** [comment]
Moving to the discussion as this is not something can be implemented for now as this makes the engine execution more complex.

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
